web/src/site-admin/SiteAdminConfigurationPage.tsx

```diff
@@ -251,6 +252,24 @@ export class SiteAdminConfigurationPage extends React.Component<Props, State> {
                                 console.error(error)
                                 this.setState({ saving: false, error })
                                 return []
+                            }),
+                            tap(() => {
+                                // Flipping the Campaigns feature flag
+                                // ("automation") requires a reload for the
+                                // Campaigns UI to be correctly rendered.
+                                //
+                                // This should be removed once the feature flag
+                                // is removed.
+                                const lastAutomationEnabled = lastConfiguration
+                                    ? (jsonc.parse(lastConfiguration.effectiveContents) as SiteConfiguration)
+                                          .experimentalFeatures?.automation
+                                    : undefined
```

Comment:
Using `&&` instead of ternary and making sure the types are both boolean (since you wouldn't want it to reload if it was `undefined` before but now `false`, because they are equivalent).

Do you need another `?` before `experimentalFeatures`?
```suggestion
                                const lastAutomationEnabled = Boolean(lastConfiguration &&  (jsonc.parse(lastConfiguration.effectiveContents) as SiteConfiguration).experimentalFeatures?.automation)
```

Reply:
> Do you need another `?` before `experimentalFeatures`?

Good catch! Thanks.

> Using `&&` instead of ternary and making sure the types are both boolean (since you wouldn't want it to reload if it was `undefined` before but now `false`, because they are equivalent).

Yeah, good point on the `"disabled"` / `undefined` equivalence.

I've pushed an update that includes the extra `?.` operators and collapses the enabled variables to `boolean`s, but didn't use `Boolean()` explicitly due to the `automation` fields being `"enabled" | "disabled" | undefined`, rather than actual booleans.

Reply:
Ah yeah sorry I forgot about that enum weirdness of this field.

Reply:
nice and clean with the `=== 'enabled'`!